apis/v1alpha1/backendpolicy_types.go

```diff
@@ -0,0 +1,232 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// BackendPolicy defines policies associated with backends such as Services.
+type BackendPolicy struct {
+	metav1.TypeMeta   `json:",inline" protobuf:"bytes,1,opt,name=typeMeta"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,2,opt,name=metadata"`
+
+	Spec   BackendPolicySpec   `json:"spec,omitempty" protobuf:"bytes,3,opt,name=spec"`
+	Status BackendPolicyStatus `json:"status,omitempty" protobuf:"bytes,4,opt,name=status"`
+}
+
+// +kubebuilder:object:root=true
+
+// BackendPolicyList contains a list of BackendPolicy
+type BackendPolicyList struct {
+	metav1.TypeMeta `json:",inline" protobuf:"bytes,1,opt,name=typeMeta"`
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,2,opt,name=metadata"`
+	Items           []BackendPolicy `json:"items" protobuf:"bytes,3,rep,name=items"`
+}
+
+// BackendPolicySpec defines desired policy for a Backend.
+type BackendPolicySpec struct {
+	// BackendRefs define which backends this policy should be applied to. If
+	// more than one BackendPolicy targets the same Backend, precedence will be
+	// given to the oldest BackendPolicy.
+	//
+	// Support: Core
+	//
+	// +optional
+	BackendRefs []BackendRef `json:"backendRefs" protobuf:"bytes,1,opt,name=backendRefs"`
+
+	// Timeouts define how timeouts should be handled for these backends.
+	//
+	// Support: Core
+	//
+	// +optional
+	Timeouts []TimeoutPolicy `json:"timeouts" protobuf:"bytes,2,rep,name=timeouts"`
+
+	// Retries define how retries should be handled for these backends.
+	//
+	// Support: Extended
+	//
+	// +optional
+	Retries []RetryPolicy `json:"retries" protobuf:"bytes,3,rep,name=retries"`
+
+	// TLS is the TLS configuration for these backends.
+	//
+	// Support: Core
```

Comment:
Shall TLS be extended feature?

Reply:
TLS on upstream connection is fairly common.
Do you have concerns about specifics of TLS in different implementations?

Reply:
I think it is probably better to make this extended. I know not every implementation supports this and those that do have varying levels of support for TLS config.